src/main/groovy/com/palantir/gradle/gitversion/BuildScanPluginInterop.groovy

```diff
@@ -0,0 +1,44 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.gitversion
+
+import java.util.function.Supplier
+import org.gradle.api.Project
+
+class BuildScanPluginInterop {
+    static void addBuildScanCustomValues(Project rootProject, Supplier<Map<String, String>> customValues) {
+        // After evaluate because while we can detect the <5.x com.gradle.build-scan project on the root project,
+        // there is no way to detect the >6.x com.gradle.enterprise settings plugins using withPlugin
```

Comment:
can't we do `project.gradle.pluginManager.withPlugin`?

Reply:
It's on `Settings`, not `Gradle`. There is no public way to get `Settings` -you have to do:

```java
Settings settings = ((org.gradle.api.internal.GradleInternal)
                        getProject().getGradle())
                .getSettings();
```

Reply:
and perhaps not write this in groovy?

Reply:
I do `gradleEnterprise.buildScan` and other lovely stuff like that in this though.